src/WebJobs.Extensions.DurableTask/LinuxAppServiceFileLogger.cs

```diff
@@ -0,0 +1,174 @@
+            this.logFilePath = Path.Combine(this.logFileDirectory, this.logFileName + ".log");
+            this.archiveFilePath = this.logFilePath + "1";
+            this.buffer = new BlockingCollection<string>(new ConcurrentQueue<string>());
+            this.currentBatch = new List<string>();
+            this.cancellationTokenSource = new CancellationTokenSource();
+
+            if (startOnCreate)
+            {
+                this.Start();
+            }
+        }
+
+        // Maximum size of individual log file in MB
+        private int MaxFileSizeMb { get; set; } = 10;
+
+        // Maximum time between successive flushes (seconds)
+        private int FlushFrequencySeconds { get; set; } = 30;
+
+        /// <summary>
+        /// TBD.
+        /// </summary>
+        /// <param name="message">TBD 1.</param>
+        public virtual void Log(string message)
+        {
+            try
+            {
+                this.buffer.Add(message);
+            }
+            catch (Exception)
+            {
+                // ignored
+            }
+        }
+
+        private void Start()
+        {
+            if (this.outputTask == null)
+            {
+                this.outputTask = Task.Factory.StartNew(this.ProcessLogQueue, null, TaskCreationOptions.LongRunning);
+            }
+        }
+
+        /// <summary>
+        /// TBD.
+        /// </summary>
+        /// <param name="timeSpan">TBD 1.</param>
+        public void Stop(TimeSpan timeSpan)
+        {
+            this.cancellationTokenSource.Cancel();
+
+            try
+            {
+                this.outputTask?.Wait(timeSpan);
+            }
+            catch (Exception)
+            {
+                // ignored
+            }
+        }
+
+        private async Task ProcessLogQueue(object state)
+        {
+            while (!this.cancellationTokenSource.IsCancellationRequested)
+            {
+                await this.InternalProcessLogQueue();
+                await Task.Delay(TimeSpan.FromSeconds(this.FlushFrequencySeconds), this.cancellationTokenSource.Token).ContinueWith(task => { });
+            }
+
+            // ReSharper disable once FunctionNeverReturns
+        }
+
+        // internal for unittests
+        internal async Task InternalProcessLogQueue()
+        {
+            string currentMessage;
+            while (this.buffer.TryTake(out currentMessage))
+            {
+                this.currentBatch.Add(currentMessage);
+            }
+
+            if (this.currentBatch.Any())
+            {
+                try
+                {
+                    await this.WriteLogs(this.currentBatch);
+                }
+                catch (Exception)
+                {
+                    // Ignored
+                }
+
+                this.currentBatch.Clear();
+            }
+        }
+
+        private async Task WriteLogs(IEnumerable<string> currentBatch)
+        {
+            Directory.CreateDirectory(this.logFileDirectory);
```

Comment:
What happens if the directory already exists?

Reply:
Nothing! The semantics of CreateDirectory are to do nothing if the path exists, no exceptions either